exchanges/okx/okx_test.go

```diff
@@ -0,0 +1,3171 @@
+		"123", enabled[0], asset.Futures)
+	if err != nil && !strings.Contains(err.Error(), "Order does not exist") {
+		t.Errorf("Okx GetOrderInfo() expecting %s, but found %v", "Order does not exist", err)
+	}
+}
+
+func TestGetDepositAddress(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.GetDepositAddress(context.Background(), currency.BTC, "", currency.USD.String()); err != nil && !errors.Is(err, errDepositAddressNotFound) {
+		t.Error("Okx GetDepositAddress() error", err)
+	}
+}
+
+func TestWithdraw(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.SkipNow()
+	}
+	withdrawCryptoRequest := withdraw.Request{
+		Exchange: ok.Name,
+		Amount:   0.00000000001,
+		Currency: currency.BTC,
+		Crypto: withdraw.CryptoRequest{
+			Address: core.BitcoinDonationAddress,
+		},
+	}
+	if _, err := ok.WithdrawCryptocurrencyFunds(context.Background(), &withdrawCryptoRequest); err != nil {
+		t.Error("Okx WithdrawCryptoCurrencyFunds() error", err)
+	}
+}
+
+func TestGetPairFromInstrumentID(t *testing.T) {
+	t.Parallel()
+	instruments := []string{
+		"BTC-USDT",
+		"BTC-USDT-SWAP",
+		"BTC-USDT-ER33234",
+	}
+	if _, err := ok.GetPairFromInstrumentID(instruments[0]); err != nil {
+		t.Error("Okx GetPairFromInstrumentID() error", err)
+	}
+	if _, ere := ok.GetPairFromInstrumentID(instruments[1]); ere != nil {
+		t.Error("Okx GetPairFromInstrumentID() error", ere)
+	}
+	if _, erf := ok.GetPairFromInstrumentID(instruments[2]); erf != nil {
+		t.Error("Okx GetPairFromInstrumentID() error", erf)
+	}
+}
+
+func TestGetActiveOrders(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	pair, err := currency.NewPairFromString("BTC-USD")
+	if err != nil {
+		t.Error(err)
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Type:      order.Limit,
+		Pairs:     currency.Pairs{pair, currency.NewPair(currency.USDT, currency.USD), currency.NewPair(currency.USD, currency.LTC)},
+		AssetType: asset.Spot,
+		Side:      order.Buy,
+	}
+	if _, err := ok.GetActiveOrders(context.Background(), &getOrdersRequest); err != nil {
+		t.Error("Okx GetActiveOrders() error", err)
+	}
+}
+
+func TestGetOrderHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Type:      order.AnyType,
+		AssetType: asset.Spot,
+		Side:      order.Buy,
+	}
+	_, err := ok.GetOrderHistory(context.Background(), &getOrdersRequest)
+	if err == nil {
+		t.Errorf("Okx GetOrderHistory() Expected: %v. received nil", err)
+	} else if err != nil && !errors.Is(err, errMissingAtLeast1CurrencyPair) {
+		t.Errorf("Okx GetOrderHistory() Expected: %v, but found %v", errMissingAtLeast1CurrencyPair, err)
+	}
+	getOrdersRequest.Pairs = []currency.Pair{
+		currency.NewPair(currency.LTC,
+			currency.BTC)}
+	if _, err := ok.GetOrderHistory(context.Background(), &getOrdersRequest); err != nil {
```

Comment:
This seems like there is a double REST request sent also:

```
Price is not being set, "fillPx":"0.2157"
AverageExecutedPrice is not being set should be: "avgPx":"0.2150655722247801"
Remaining amount is not being set correctly should be "accFillSz":"929.948935" - "fillSz":"743.609809" or if filled 0
Executed amount is not being set "fillSz":"743.609809" or if filled "accFillSz":"929.948935"
Amount is not being set correctly should be "accFillSz":"929.948935"
Cost is not being set should be "sz":"200"
```

Reply:
I couldn't test that because the API Key I has no orders to fetch. But, I have added a simple length check to determine if there are other orders left to retrieve.